src/librbd/mirror/snapshot/PromoteRequest.h

```diff
@@ -83,6 +83,7 @@ class PromoteRequest {
   Context *m_on_finish;
 
   uint64_t m_rollback_snap_id = CEPH_NOSNAP;
+  bool m_requires_orphan = false;
```

Comment:
nit: Is setting a value here needed? As `m_requires_orphan` is passed to `util::can_create_primary_snapshot` and has this section in the beginning of the `can_create_primary_snapshot` function:
```
  if (requires_orphan != nullptr) {
    *requires_orphan = false;
  }
```
IIUC it will set requires_orphan to false no matter its value as long as it's not nullptr.

Reply:
Do we really needs `requires_orphans` as a member variable? I don't see how it is used outside the `send` method.

Reply:
@trociny could be avoided, just for brevity, since we are using `can_create_primary_snapshot` before rollback() call